pkg/distributor/field_detection.go

```diff
@@ -316,21 +316,30 @@ func isJSON(line string) bool {
 
 func detectLevelFromLogLine(log string) string {
 	if strings.Contains(log, "info:") || strings.Contains(log, "INFO:") ||
+		strings.Contains(log, "[info]") || strings.Contains(log, "[INFO]") ||
 		strings.Contains(log, "info") || strings.Contains(log, "INFO") {
 		return constants.LogLevelInfo
 	}
 	if strings.Contains(log, "err:") || strings.Contains(log, "ERR:") ||
+		strings.Contains(log, "[err]") || strings.Contains(log, "[ERR]") ||
+		strings.Contains(log, "[error]") || strings.Contains(log, "[ERROR]") ||
 		strings.Contains(log, "error") || strings.Contains(log, "ERROR") {
 		return constants.LogLevelError
 	}
 	if strings.Contains(log, "warn:") || strings.Contains(log, "WARN:") ||
+		strings.Contains(log, "[warn]") || strings.Contains(log, "[WARN]") ||
+		strings.Contains(log, "[warning]") || strings.Contains(log, "[WARNING]") ||
 		strings.Contains(log, "warning") || strings.Contains(log, "WARNING") {
 		return constants.LogLevelWarn
 	}
-	if strings.Contains(log, "CRITICAL:") || strings.Contains(log, "critical:") {
+	if strings.Contains(log, "CRITICAL:") || strings.Contains(log, "critical:") ||
+		strings.Contains(log, "[critical]") || strings.Contains(log, "[CRITICAL]") ||
+		strings.Contains(log, "critical") || strings.Contains(log, "CRITICAL") {
```

Comment:
these are the lines, without any special characters around them, that are likely to cause false positives. however, since I see we already have that pattern in here we can merge and wait to see if we get a bug report for false positives, at which point I recommend removing all the ones without special characters around them.

Reply:
I made the changes for `critical` and `error` try to avoid false positive for both.